public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilder.tsx

```diff
@@ -42,15 +43,94 @@ export const PromQueryBuilder = React.memo<Props>((props) => {
   const { datasource, query, onChange, onRunQuery, data, showExplain } = props;
   const [highlightedOp, setHighlightedOp] = useState<QueryBuilderOperation | undefined>();
   const [showDrawer, setShowDrawer] = useState<boolean>(false);
+  const [llmAppEnabled, updateLlmAppEnabled] = useState<boolean>(false);
 
   const lang = { grammar: promqlGrammar, name: 'promql' };
 
   const initHints = datasource.getInitHints();
 
+  useEffect(() => {
+    async function checkLlms() {
+      const check = await isLLMPluginEnabled();
+      updateLlmAppEnabled(check);
+    }
+    checkLlms();
+  }, []);
+
+  const theme = useTheme2();
+  const styles = getStyles(theme);
+
+  const queryAssistantButton = function () {
+    const llmAppDisabled = !llmAppEnabled;
+    const noMetricSelected = !query.metric;
+
+    const button = () => {
```

Comment:
would it make sense to refactor this button into a new component?

Reply:
Yes it would!